File: src/main/java/cope/nebula/util/world/damage/DamageUtil.java
package cope.nebula.util.world.damage;

import cope.nebula.util.Globals;
import net.minecraft.util.CombatRules;
import net.minecraft.world.World;

/**
 * Calculates general damages
 *
 * @author aesthetical
 * @since 3/19/22
 */
public class DamageUtil implements Globals {
    /**
     * Gets the scaled damage based off of the difficulty
     * @param damage
     * @return
     */
    public static float getScaledDamage(float damage) {
        World world = mc.world;
        if (world == null) {
            return damage;
        }

        int diff = world.getDifficulty().getId();
        return damage * ((float) diff * 0.5f);
    }
}

File: src/main/kotlin/wtf/nebula/util/combat/DamageUtil.kt
package wtf.nebula.util.combat

import net.minecraft.world.EnumDifficulty
import wtf.nebula.util.Globals
import kotlin.math.min

object DamageUtil : Globals {
    fun getDamageMultiplierOnDiff(damage: Float): Float {
        return when (mc.world.difficulty) {
            EnumDifficulty.PEACEFUL -> 0.0f
            EnumDifficulty.EASY -> min(damage / 2.0f + 1.0f, damage)
            EnumDifficulty.HARD -> damage * 3.0f / 2.0f
            else -> damage
        }
    }
}